Vue Vite Docker WSL2. Internal server error: EACCES: permission denied, stat '/root/.node_modules/plugin-vue:export-helper'

### Describe the bug

Error running dev server in WSL2 Docker.

docker-compose:
```
  frontend:
    build: .
    command: npm run dev
    volumes:
      - ./:/frontend
    ports:
      - '8080:8080'
```

Dockerfile:
```
FROM node:lts-alpine
WORKDIR /frontend
COPY package*.json ./
RUN npm install
COPY . .
CMD ["npm", "run", "dev"]
```

vite.config.js:
```
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue()],
  server: {
    host: true,
    port: 8080
  }
})
```

![screenshot4](https://user-images.githubusercontent.com/13174541/152646445-78430751-2337-4c24-91fc-16bc1f83e83c.png)

### Reproduction

https://github.com/sshiling/Vite-Vue-Docker-WSL2-Bug

### System Info

```shell
System:
    OS: Linux 4.19 Ubuntu 20.04 LTS (Focal Fossa)        
    CPU: (1) x64 Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz
    Memory: 129.67 MB / 979.44 MB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 6.14.15 - /usr/bin/npm
```


### Used Package Manager

npm

### Logs

```shell
frontend_1  |   vite v2.7.13 dev server running at:
frontend_1  |
frontend_1  |   > Local:    http://localhost:8080/
frontend_1  |   > Network:  http://172.19.0.2:8080/
frontend_1  |
frontend_1  |   ready in 259ms.
frontend_1  |
frontend_1  | 2:31:01 PM [vite] Internal server error: EACCES: permission denied, stat '/root/.node_modules/plugin-vue:export-helper'
frontend_1  |   Plugin: vite:import-analysis
frontend_1  |   File: /frontend/src/App.vue
frontend_1  |       at Object.statSync (node:fs:1536:3)
frontend_1  |       at isDirectory (/frontend/node_modules/resolve/lib/sync.js:31:23)
frontend_1  |       at loadNodeModulesSync (/frontend/node_modules/resolve/lib/sync.js:200:17)
frontend_1  |       at Function.resolveSync [as sync] (/frontend/node_modules/resolve/lib/sync.js:107:17)
frontend_1  |       at resolveFrom$3 (/frontend/node_modules/vite/dist/node/chunks/dep-f5552faa.js:3911:29)
frontend_1  |       at resolvePackageData (/frontend/node_modules/vite/dist/node/chunks/dep-f5552faa.js:30046:19)
frontend_1  |       at /frontend/node_modules/vite/dist/node/chunks/dep-f5552faa.js:30426:15
frontend_1  |       at Array.find (<anonymous>)
frontend_1  |       at tryNodeResolve (/frontend/node_modules/vite/dist/node/chunks/dep-f5552faa.js:30425:44)
frontend_1  |       at Context.resolveId (/frontend/node_modules/vite/dist/node/chunks/dep-f5552faa.js:30257:28)
```


### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

This is happening to me as well.
But I'm on PopOs, i don't think this is something to do with WSL.
```bash
frontend  | error when starting dev server:
frontend  | Error: Build failed with 11 errors:
frontend  | node_modules/.pnpm/resolve@1.22.0/node_modules/resolve/lib/sync.js:31:22: error: [plugin: vite:dep-scan] EACCES: permission denied, stat '/root/.node_modules/@heroicons/react/outline/SearchIcon'
frontend  | node_modules/.pnpm/resolve@1.22.0/node_modules/resolve/lib/sync.js:31:22: error: [plugin: vite:dep-scan] EACCES: permission denied, stat '/root/.node_modules/@heroicons/react/outline/SearchIcon'
frontend  | node_modules/.pnpm/resolve@1.22.0/node_modules/resolve/lib/sync.js:31:22: error: [plugin: vite:dep-scan] EACCES: permission denied, stat '/root/.node_modules/@heroicons/react/outline/SearchIcon'
frontend  | node_modules/.pnpm/resolve@1.22.0/node_modules/resolve/lib/sync.js:31:22: error: [plugin: vite:dep-scan] EACCES: permission denied, stat '/root/.node_modules/dayjs/plugin/relativeTime'
frontend  | node_modules/.pnpm/resolve@1.22.0/node_modules/resolve/lib/sync.js:31:22: error: [plugin: vite:dep-scan] EACCES: permission denied, stat '/root/.node_modules/dayjs/plugin/relativeTime'
frontend  | ...
frontend  |     at failureErrorWithLog (/app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:1493:15)
frontend  |     at /app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:1151:28
frontend  |     at runOnEndCallbacks (/app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:941:63)
frontend  |     at buildResponseToResult (/app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:1149:7)
frontend  |     at /app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:1258:14
frontend  |     at /app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:629:9
frontend  |     at handleIncomingPacket (/app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:726:9)
frontend  |     at Socket.readFromStdout (/app/client/node_modules/.pnpm/esbuild@0.13.15/node_modules/esbuild/lib/main.js:596:7)
frontend  |     at Socket.emit (node:events:390:28)
frontend  |     at addChunk (node:internal/streams/readable:315:12)
frontend exited with code 1
```

--- a/packages/vite/src/node/utils.ts
+++ b/packages/vite/src/node/utils.ts
@@ -83,6 +83,7 @@ export function resolveFrom(
 ): string {
   return resolve.sync(id, {
     basedir,
+    paths: [],
     extensions: ssr ? ssrExtensions : DEFAULT_EXTENSIONS,
     // necessary to work with pnpm
     preserveSymlinks: preserveSymlinks || isRunningWithYarnPnp || false

